datumaro/components/validator.py

```diff
@@ -876,14 +912,244 @@ def generate_reports(self, stats):
         return reports
 
 
+class SegmentationValidator(_Validator):
+    """
+    A validator class for (instance) segmentation tasks.
+    """
+
+    def __init__(self):
+        super().__init__(TaskType.segmentation)
+
+    def compute_statistics(self, dataset):
+        """
+        Computes statistics of the dataset for the segmentation task.
+
+        Parameters
+        ----------
+        dataset : IDataset object
+
+        Returns
+        -------
+        stats (dict): A dict object containing statistics of the dataset.
+        """
+
+        stats, filtered_anns = self._compute_common_statistics(dataset)
+
+        # segmentation-specific
+        mask_template = {
+            'area': deepcopy(self.numerical_stat_template),
+            'width': deepcopy(self.numerical_stat_template),
+            'height': deepcopy(self.numerical_stat_template)
+        }
+
+        stats['items_with_invalid_value'] = {}
+        stats['mask_distribution_in_label'] = {}
+        stats['mask_distribution_in_attribute'] = {}
+        stats['mask_distribution_in_dataset_item'] = {}
+
+        dist_by_label = stats['mask_distribution_in_label']
+        dist_by_attr = stats['mask_distribution_in_attribute']
+        mask_dist_in_item = stats['mask_distribution_in_dataset_item']
+        items_w_invalid_val = stats['items_with_invalid_value']
+
+        def _generate_ann_mask_info(area, _w, _h):
+            return {
+                'area': area,
+                'width': _w,
+                'height': _h,
+            }
+
+        def _update_mask_stats_by_label(item_key, ann, mask_label_stats):
+            mask_has_error = False
+
+            _x, _y, _w, _h = ann.get_bbox()
+            if ann.type == AnnotationType.mask:  # temp.
```

Comment:
Can you make the comment more clear?

Reply:
Once #226 is resolved, this code can be deleted.
I'll leave this as a comment.

Reply:
Ok. I don't think this fix completely fixes the problem, though, because when we have `[0, 0, 0]`, `[0, 1, 0]` the fix will always return 1.

Reply:
In that case, the bounding box should be 1x1, but currently, it returns 0x0.
```
>>> mask = np.array([[0,0,0], [0,1,0]])
>>> find_mask_bbox(mask)
[1, 1, 0, 0]
```
So, I think this temporal code can fix the problem.

Reply:
No, I meant that two cases: `[0, 0, 0]` and `[0, 1, 0]` - both will return 1?